Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -292,8 +298,19 @@ void MediaPlayerPrivateGStreamerMSE::sourceSetup(GstElement* sourceElement)
 
 void MediaPlayerPrivateGStreamerMSE::updateStates()
 {
-    bool shouldBePlaying = !m_isPaused && readyState() >= MediaPlayer::ReadyState::HaveFutureData;
+    bool shouldBePlaying = !m_isPaused && !m_isPlaybackRatePaused && readyState() >= MediaPlayer::ReadyState::HaveFutureData;
     GST_DEBUG_OBJECT(pipeline(), "shouldBePlaying = %s, m_isPipelinePlaying = %s", boolForPrinting(shouldBePlaying), boolForPrinting(m_isPipelinePlaying));
+
+    GstState state, pending;
+    GstStateChangeReturn getStateResult = gst_element_get_state(pipeline(), &state, &pending, 0);
+    if (getStateResult == GST_STATE_CHANGE_ASYNC) {
+        // Changing pipeline state during async state change will be either rejected by changePipelineState()
+        // or ignored by playbin (overwritten by the async tartet state after transitions completes).
+        // Let's wait until it's done and then apply play/pause if needed.
```

Comment:
This if block makes me a bit uneasy.

> Changing pipeline state during async state change will be either rejected by changePipelineState()

Why? Is that behavior correct?

> or ignored by playbin (overwritten by the async tartet state after transitions completes).

Same questions (why and whether that behavior is correct, a bug of GStreamer or has some sort of rationale I'm missing).

Reply:
I don't really know, never investigated the issue myself.

Reply:
Could you test what happens if this if is removed?

Reply:
Nothing happens if it's removed, at least upstream.